src/counterfactuals/core_struct.jl

```diff
@@ -92,31 +57,27 @@ function CounterfactualExplanation(
         data,
         M,
         deepcopy(generator),
-        generative_model_params,
-        params,
         nothing,
         convergence,
         num_counterfactuals,
         initialization,
     )
 
-    # Initialization:
-    adjust_shape!(ce)                                           # adjust shape to specified number of counterfactuals
-    ce.s′ = encode_state(ce)            # encode the counterfactual state
-    ce.s′ = initialize_state(ce)        # initialize the counterfactual state
-    ce.x′ = decode_state(ce)            # decode the counterfactual state
-
     # Initialize search:
     ce.search = Dict(
         :iteration_count => 0,
-        :times_changed_features => zeros(size(decode_state(ce))),
-        :path => [ce.s′],
+        :mutability => DataPreprocessing.mutability_constraints(data),
+        :potential_neighbours => find_potential_neighbours(ce),
     )
 
-    # This is lifted out of the above ce.search initialization because calling converged(ce) might self-reference
-    # the above fields, which are not yet initialized.
-    ce.search[:converged] = Convergence.converged(ce.convergence, ce)
-    ce.search[:terminated] = terminated(ce)
+    # Initialization:
+    adjust_shape!(ce)                   # adjust shape to specified number of counterfactuals
+    ce.s′ = encode_state(ce)            # encode the counterfactual state
+    ce.s′ = initialize_state(ce)        # initialize the counterfactual state
+    ce.x′ = decode_state(ce)            # decode the counterfactual state
+
+    ce.search[:path] = [ce.s′]
+    ce.search[:times_changed_features] = zeros(size(decode_state(ce)))
```

Comment:
The lines 79 and 80 are not that elegant, but the `:path` and `:times_changed_features` fields need to be set after running the initialization methods, while the `:mutability` field has to be set before running the initialization methods, so this solution seems necessary - let me know if you think there's a better solution.